pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentSizeBasedFlushThresholdUpdater.java

```diff
@@ -102,8 +106,14 @@ public synchronized void updateFlushThreshold(PartitionLevelStreamConfig streamC
     // less same characteristics at any one point in time).
     // However, when we start a new table or change controller mastership, we can have any partition completing first.
     // It is best to learn the ratio as quickly as we can, so we allow any partition to supply the value.
-    // FIXME: The stream may not have partition "0"
-    if (new LLCSegmentName(newSegmentName).getPartitionGroupId() == 0 || _latestSegmentRowsToSizeRatio == 0) {
+
+    // Partition group id 0 might not be available always. We take the smallest available partition id in that case to update the threshold
+    int smallestAvailablePartitionGroupId =
+        partitionGroupMetadataList.stream().min(Comparator.comparingInt(PartitionGroupMetadata::getPartitionGroupId))
+            .map(PartitionGroupMetadata::getPartitionGroupId).orElseGet(() -> 0);
```

Comment:
seems like the `orElseGet(()->0)` is added only for the test purposes? In an actual setup, you would never reach that path? In that case, can we just send the right input in the test, instead of doing this?
We still might have to guard against the case where all shards expired (can happen in kinesis). But then, we can set some special number here (-1) instead.
wdyt?

Reply:
Also, it looks like as long as there are non-zero partitions in the metadata list, we should get some min or another. So, the case of no partitions being available in metadata list can potentially be handled. Better yet, a pre-condition that it is not empty. Speaking of which, it is possible that it is empty? Say, one partition got started, and the administrator closed it for some reason. So, the current list of partitions is empty. Possible, right? if so, then we should update the ratio (with whatever partition ID is completing)